### PR TITLE
New package: monophony-2.10.0 and dependency

### PR DESCRIPTION
**Pull request by Bnyro**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc
- I built this PR locally for these architectures:
  - i686

### CODE REVIEW

File: srcpkgs/monophony/template [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'monophony'
+pkgname=monophony
+version=2.2.0
+revision=1
+build_style=gnu-makefile
+hostmakedepends="python3-Nuitka"
+depends="libadwaita python3-gobject python3-requests python3-ytmusicapi python3-mpris_server yt-dlp"
```
Comment (zehkira) [inline srcpkgs/monophony/template:7]: Monophony doesn't use requests anymore (since v1.10.0).

Comment (Bnyro) [reply srcpkgs/monophony/template]: neat, thanks

**Comment by Bnyro** [comment]
Hey @zehkira, what's the state of Python3.12 for Monophony?

When compiling, I get
```
Nuitka-Reports:INFO: Compilation report written to file 'nuitka-crash-report.xml'.
Traceback (most recent call last):
  File "/usr/lib/python3.12/site-packages/nuitka/__main__.py", line 155, in <module>
    main()
  File "/usr/lib/python3.12/site-packages/nuitka/__main__.py", line 141, in main
    MainControl.main()
  File "/usr/lib/python3.12/site-packages/nuitka/MainControl.py", line 1093, in main
    _main()
  File "/usr/lib/python3.12/site-packages/nuitka/MainControl.py", line 962, in _main
    result, options = compileTree()
                      ^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/MainControl.py", line 836, in compileTree
    makeSourceDirectory()
  File "/usr/lib/python3.12/site-packages/nuitka/MainControl.py", line 437, in makeSourceDirectory
    source_code = CodeGeneration.generateModuleCode(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/CodeGeneration.py", line 578, in generateModuleCode
    return _generateModuleCode(module=module, data_filename=data_filename)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/CodeGeneration.py", line 538, in _generateModuleCode
    function_code, function_decl = generateFunctionBodyCode(
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/CodeGeneration.py", line 482, in generateFunctionBodyCode
    function_code = getFunctionCode(
                    ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/FunctionCodes.py", line 612, in getFunctionCode
    return _getFunctionCode(
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/FunctionCodes.py", line 652, in _getFunctionCode
    generateStatementSequenceCode(
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/CodeHelpers.py", line 241, in generateStatementSequenceCode
    _generateStatementSequenceCode(
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/CodeHelpers.py", line 230, in _generateStatementSequenceCode
    generateStatementCode(
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/CodeHelpers.py", line 190, in generateStatementCode
    statement_dispatch_dict[statement.kind](
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/TryCodes.py", line 86, in generateTryCode
    generateStatementSequenceCode(
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/CodeHelpers.py", line 241, in generateStatementSequenceCode
    _generateStatementSequenceCode(
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/CodeHelpers.py", line 225, in _generateStatementSequenceCode
    generateStatementsFrameCode(
    getFrameGuardHeavyCode(
  File "/usr/lib/python3.12/site-packages/nuitka/code_generation/FrameCodes.py", line 311, in getFrameGuardHeavyCode
    renderTemplateFromString(
  File "/usr/lib/python3.12/site-packages/nuitka/utils/Jinja2.py", line 140, in renderTemplateFromString
    _template_cache[template_str] = getTemplateFromString(template_str)
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/utils/Jinja2.py", line 129, in getTemplateFromString
    return getEnvironment(
           ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/utils/Jinja2.py", line 77, in getEnvironment
    jinja2 = getJinja2Package()
             ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/utils/Jinja2.py", line 62, in getJinja2Package
    _loaded_pkg_resources = importFromInlineCopy("pkg_resources", must_exist=False)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/utils/Importing.py", line 184, in importFromInlineCopy
    return _importFromFolder(
           ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/utils/Importing.py", line 147, in _importFromFolder
    return __import__(module_name, level=0)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/nuitka/build/inline_copy/pkg_resources/pkg_resources/__init__.py", line 2191, in <module>
    register_finder(pkgutil.ImpImporter, find_on_path)
                    ^^^^^^^^^^^^^^^^^^^
AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?
```

**Comment by zehkira** [comment]
Nuitka does not support Python 3.12 yet.

**Comment by Bnyro** [comment]
I assume there's no way to use monophony without compiling via Nuitka currently? It would be handy if it could be installed like any other Python module instead of being compiled first.

**Comment by zehkira** [comment]
You could patch the `Makefile` and the `pyproject.toml` to install it as a module. I don't recommend doing that though.

**Comment by Bnyro** [comment]
Okay, we'll wait for Nuitka to support Py3.12 then. Thanks for the answers!

**Comment by ahesford** [comment]
As discussed in https://github.com/void-linux/void-packages/issues/47982, nuitka does not appear to be suitable for packaging on Void.

**Comment by Bnyro** [comment]
Closing for the given reasons.

**Comment by zehkira** [comment]
Starting with the upcoming v2.10.0 release, Monophony will no longer use Nuitka.

**Comment by Bnyro** [comment]
That's awesome, I'll give this PR an other shot once it's released 🚀

**Pull request closed without merging**